### DISCUSSION TITLE
How can I build an executable file using a built python script?

### DISCUSSION THREAD

**Discussion by PatchyPhoenix** [pyinstaller/pyinstaller#7226] [Help] [answered]
I want to make an executable made in python, that can build other python scripts to executables.
When I tried to build an executable from another executable, it stopped at `Extending PYTHONPATH with paths` and the script restarted.

**Comment by rokm** [comment, marked as answer]
You cannot do that with PyInstaller; the frozen executable does not provide a full python environment that PyInstaller itself expects to find when building an executable.

**Reply by PatchyPhoenix** [reply]
Ok.
Is there any other work around I can try to build an executable without having python involved?

**Reply by rokm** [reply]
Not that I know of. 

But think about it this way: if you want to build an executable from an arbitrary python script, you need to have all its dependencies available. When you build a builder application, that bundle includes only a subset of python stdlib and dependencies that it itself requires. So consequently, it cannot be used to freeze an arbitrary python script, unless the script's dependencies represent the subset of dependencies of the builder application. (And that's ignoring the way PyInstaller spawns subprocesses, which is incompatible with PyInstaller-frozen applications).

**Reply by PatchyPhoenix** [reply]
Ok, thank you
